src/parser.rs

```diff
@@ -290,6 +290,46 @@ impl<'a> Parser<'a> {
         }
     }
 
+    // Parse an array index as a natural number
+    fn array_index(&mut self) -> Option<String> {
```

Comment:
I think parsing integers accepts the +xx and -xx notation, which I don't think should be applicable. I'll need to look through the parser methods...